### PR TITLE
Add control-plane documentation

### LINKED ISSUE #87: Update concepts docs with "what we're building"

### PR DESCRIPTION
**Pull request by AaronCrawfis**
Adds documentation on the Project Radius runtime services

### CODE REVIEW

File: docs/content/concepts/runtime/index.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+<img src="runtime-services.png" alt="Diagram of the Project Radius runtime services" width="700" />
+
+## Services
+
+| Service | Description | Self-hosted | Microsoft Azure |
+|---------|-------------|-------------|-----------------|
+| **Universal control plane (UCP)/<br />Azure Resource Manager (ARM)** | The "front door" to the runtime services. Provides authentication, authorization, and other central capabilities. | Self-hosted UCP service | Azure Resource Manager (ARM) |
+| **Radius resource provider (RP)** | Manages Radius applications and app-model resources. | Self-hosted RP service | Central `Applications.*` Azure resource provider |
+| **Deployment engine (DE)** | Orchestrates UCP and ARM deployments when Bicep or ARM JSON is used to deploy. | Self-hosted DE service | Azure Resource Manager (ARM) |
+| **Bicep language extensibility provider (BEP)** | Handles Radius applications and app-model resources when using the Bicep language to deploy. | Self-hosted BEP service | Central BEP service |
```
Comment (jkotalik) [inline docs/content/concepts/runtime/index.md]: No more Bicep extensibility provider 👍

Comment (jkotalik) [reply docs/content/concepts/runtime/index.md]: I would not have it for either diagram for now (even if our goal state is to potentially have it).

File: docs/content/concepts/runtime/index.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+The Project Radius runtime is a set of services that accepts a Radius application definition and deploys it to the target [platform and runtime]({{< ref platforms >}}).
+
+<img src="runtime-services.png" alt="Diagram of the Project Radius runtime services" width="700" />
+
+## Services
+
+| Service | Description | Self-hosted | Microsoft Azure |
+|---------|-------------|-------------|-----------------|
+| **Universal control plane (UCP)/<br />Azure Resource Manager (ARM)** | The "front door" to the runtime services. Provides authentication, authorization, and other central capabilities. | Self-hosted UCP service | Azure Resource Manager (ARM) |
```
Comment (jkotalik) [inline docs/content/concepts/runtime/index.md]: I'd mention _routing_ as a key part.

File: docs/content/concepts/runtime/index.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+| **Radius resource provider (RP)** | Manages Radius applications and app-model resources. | Self-hosted RP service | Central `Applications.*` Azure resource provider |
+| **Deployment engine (DE)** | Orchestrates UCP and ARM deployments when Bicep or ARM JSON is used to deploy. | Self-hosted DE service | Azure Resource Manager (ARM) |
+| **Bicep language extensibility provider (BEP)** | Handles Radius applications and app-model resources when using the Bicep language to deploy. | Self-hosted BEP service | Central BEP service |
+
+### Universal control plane
+
+The universal control plane (UCP) provides the front-door endpoint, authentication, authorization, and other central capabilities for a self-hosted Radius environment. When deploying to a Radius environment using a deployment JSON file (Bicep) or through sinbgle imperative commands (CLI, Terraform), UCP handles the request and sends it to the appropriate plane and resource provider.
```
Comment (jkotalik) [inline docs/content/concepts/runtime/index.md]:
```suggestion
The universal control plane (UCP) provides the front-door endpoint, authentication, authorization, and other central capabilities for a self-hosted Radius environment. When deploying to a Radius environment using a deployment JSON file (Bicep) or through single imperative commands (CLI, Terraform), UCP handles the request and sends it to the appropriate plane.
```

File: docs/content/concepts/runtime/index.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+- The RP handles CRUDL operations for Radius applications, connectors, and other app-model resources
+
+<img src="ucp-selfhosted.png" alt="Diagram of the UCP on a self-hosted environment with template deployment" width="1000" />
+{{% /codetab %}}
+
+{{% codetab %}}
+When using Terraform or the rad CLI to deploy and interact with Radius applications:
+
+- UCP routes the CRUDL requests directly to the Radius RP.
+- The client (CLI, Terraform) acts as the DE in this case, bypassing the built-in DE and BEP.
```
Comment (jkotalik) [inline docs/content/concepts/runtime/index.md]: I think having a separate diagram for terraform may be better if we are going to document it.

File: docs/content/concepts/runtime/index.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+<img src="azure-deployment.png" alt="Diagram of the Azure environment with template deployment" width="1000" />
+{{% /codetab %}}
+
+{{% codetab %}}
+In Microsoft Azure environments, Azure Resource Manager (ARM) is used instead of UCP.
+
+- ARM routes the CRUDL requests directly to the Radius RP.
+- The client (CLI, Terraform) acts as the DE in this case, bypassing the built-in DE and BEP.
+
+<img src="azure-imperative.png" alt="Diagram of the Azure environment with imperative deployment" width="1000" />
```
Comment (jkotalik) [inline docs/content/concepts/runtime/index.md]: The term "imperative" is a bit confusing to me, does that refer to Terraform deployments?

File: docs/content/concepts/runtime/index.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+### Radius resource provider
+
+The Radius resource provider handles CRUDL requests for Radius applications, connectors, and other app-model resources. The resource provider is split into two namespaces:
```
Comment (jkotalik) [inline docs/content/concepts/runtime/index.md]:
```suggestion
The Radius resource provider handles CRUDL requests for Radius applications, connectors, and other app-model resources. The resource provider handles:
```

Are providers called namespaces in Azure?

File: docs/content/concepts/runtime/index.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+- **Applications.Core**: The _core_ resources of an application, including the application itself, containers, gateways, and routes.
+- **Applications.Connectors**: The _connector_ resources that can be used within an application.
+
+Backing resources may be created deleted by the Radius RP for some resources. For example, a Radius container creates a backing Pod or Container App on the target platform and container runtime.
+
+### Deployment engine
+
+The DE service is used when Bicep or ARM JSON is used to deploy Radius applications. It builds the dependency graph of resources and orchestrates the CRUDL requests to the respective resource providers.
```
Comment (jkotalik) [inline docs/content/concepts/runtime/index.md]:
```suggestion
The DE service is used when Bicep or ARM JSON is used to deploy Radius applications. It parses and builds the dependency graph of resources and orchestrates the CRUDL requests to the respective resource providers.
```

File: docs/content/concepts/runtime/index.md (hunk 6 above)
Comment (AaronCrawfis) [reply docs/content/concepts/runtime/index.md]: They're separate, but they map 1-1, where a namespace maps to a resource provider. Used the term namespace to focus more on what the user writes and the classes of resources, vs. how they are implemented.

File: docs/content/concepts/runtime/index.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
+---
+type: docs
+title: Project Radius runtime services
```
Comment (rynowak) [inline docs/content/concepts/runtime/index.md]: 🚨🚨 🚨 🚨 🚨 🚨 🚨 🚨 🚨 🚨 🚨 🚨 🚨 

Can we pick a better name than **runtime** for this? None of these are runtimes.  They don't play a role in how your code functions or interact with your code directly. 

🚨🚨 🚨 🚨 🚨 🚨 🚨 🚨 🚨 🚨 🚨 🚨 🚨

Comment (AaronCrawfis) [reply docs/content/concepts/runtime/index.md]: Yeah I'm good with that. Couple ideas:

- Orchestrator services
- Control-plane services
- Management services

File: docs/content/concepts/runtime/index.md (hunk 1 above)
Comment (AaronCrawfis) [reply docs/content/concepts/runtime/index.md]: Removed

File: docs/content/concepts/runtime/index.md (hunk 8 above)
Comment (rynowak) [reply docs/content/concepts/runtime/index.md]: I'd lean towards control-plane or management services, but I think any of them are acceptable. What do others think?

**Pull request merged**